### PR TITLE
add privileged config option

### PR DESCRIPTION
**Pull request by mfleader**
## Changes introduced with this PR

Add [podman privileged](https://docs.podman.io/en/latest/markdown/podman-run.1.html#privileged) as a config option for the deployer. 

[Privileged container execution reference](https://www.redhat.com/sysadmin/privileged-flag-container-engines).

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by dbutenhof — COMMENTED** [review]
Doesn't this have to be translated to `--privileged` somewhere for the `podman` command?

And any intent to do this for the docker deployer as well? (Not that I'm complaining: everyone should use podman anyway. 😆 )

**Comment by mfleader** [comment]
> Doesn't this have to be translated to `--privileged` somewhere for the `podman` command?
> 
> And any intent to do this for the docker deployer as well? (Not that I'm complaining: everyone should use podman anyway. 😆 )

Afaik, no, but I will verify more completely. I've never seen any program where you have to enable the _use_ of a CLI option through the configuration of another file. Usually, they are just different entrypoints with different levels of precedence to alter same the behavior.

**Comment by mfleader** [comment]
> Doesn't this have to be translated to `--privileged` somewhere for the `podman` command?
> 
> And any intent to do this for the docker deployer as well? (Not that I'm complaining: everyone should use podman anyway. 😆 )

Okay yeah I did need to implement the behavior the argsbuilder interface and call it in `Deploy()`. I thought we generated our own container runtime toml config, as in [containers.confg](https://github.com/containers/common/blob/main/docs/containers.conf.5.md).

File: schema.go [hunk 1]
```diff
@@ -320,6 +320,16 @@ var Schema = schema.NewTypedScopeSchema[*Config](
 				nil,
 				nil,
 			).Disable(notImplemented),
+			"Privileged": schema.NewPropertySchema(
+				schema.NewBoolSchema(),
+				schema.NewDisplayValue(schema.PointerTo("Privileged"), schema.PointerTo("Execute container process without any further \"security\" lockdown"), nil),
```
Comment (webbnh) [inline schema.go:325]: I'd recommend borrowing [Podman's text:](https://docs.podman.io/en/latest/markdown/podman-run.1.html#privileged)
> "Execute container process without security features that isolate the container from the host"

**Review by webbnh — APPROVED** [review]
Looks good; one suggestion.

**Review by dbutenhof — APPROVED** [review]

File: schema.go (hunk 1 above)
Comment (mfleader) [reply schema.go:325]: did it

Comment (webbnh) [reply schema.go]: I eventually found that you'd lifted your original text from their blog post, so...yeah.  (But, I didn't feel like "lockdown" and the scare-quotes around "security" were the best for our documentation.)

**Pull request merged**